perffile/events.go

```diff
@@ -75,30 +79,37 @@ func (e eventUnknown) Generic() EventGeneric {
 }
 
 // EventHardware represents a hardware event.
+type EventHardware struct {
+	ID        EventHardwareID
+	PMUTypeID uint32
+}
+
+func (e EventHardware) Generic() EventGeneric {
+	id := uint64(e.ID) | uint64(e.PMUTypeID)<<32
```

Comment:
Maybe panic if e.ID has bits outside 0xff set? At minimum, they should be masked off.

Reply:
No longer relevant because ID is uint8.